src/inmanta/ast/statements/assign.py

```diff
@@ -555,16 +555,36 @@ def __repr__(self) -> str:
         )
 
 
-class StringFormat(ReferenceStatement):
+class FormattedString(ReferenceStatement):
     """
-    Create a new string by doing a string interpolation
+    This class is an abstraction around a string containing references to variables.
     """
 
     __slots__ = ("_format_string", "_variables")
 
-    def __init__(self, format_string: str, variables: typing.List[typing.Tuple["Reference", str]]) -> None:
-        ReferenceStatement.__init__(self, [k for (k, _) in variables])
+    def __init__(self, format_string: str, variables: Sequence["Reference"]) -> None:
+        super().__init__(variables)
         self._format_string = format_string
+
+    def execute(self, requires: typing.Dict[object, object], resolver: Resolver, queue: QueueScheduler) -> object:
+        return super().execute(requires, resolver, queue)
+
```

Comment:
```suggestion
```

I don't know if it adds any value to have execute here? (definitely not if there is no documentation on it)